src/bayesdb.py

```diff
@@ -377,6 +377,19 @@ def changes(self):
         """
         return self._sqlite3.changes()
 
+    def dump_models(self, population_name, path):
+        """Write information about a population's models to a file, as JSON.
+
+        The intent is that this information, combined with the data
+        table (not itself dumped here), should be sufficient to simulate 
+        any of the models.
+        """
+        import json
+        backend = self.backends['cgpm']
```

Comment:
Hard-coding the backend to `'cgpm'` isn't correct here since the backend of the population could be `'loom'`, instead we should use `core.bayesdb_generator_backend` to obtain the backend.  If the `'loom'` backend does not implement `json_ready_models` the user will receive a `NotImplementedError()`.

Reply:
So the caller becomes responsible for providing the generator_id (required for the call to core.bayesdb_generator_backend)?  The reason I did it this way was because I couldn't figure out where a generator id would come from.  (And I couldn't find an explanation of what a generator was, in the first place.)

Reply:
Yes correct.